src/main/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource.java

```diff
@@ -147,19 +148,31 @@
                 .orElse(new ResponseEntity<>(HttpStatus.NOT_FOUND));
     }
 
-    @ApiOperation(value = "Return the details for an offender", tags = "-- Popular core APIs --")
+    @ApiOperation(value = "Return the details for an offender. If multiple offenders found the active one wll be returned", tags = "-- Popular core APIs --")
     @ApiResponses(
-            value = {
-                    @ApiResponse(code = 400, message = "Invalid request", response = ErrorResponse.class),
-                    @ApiResponse(code = 500, message = "Unrecoverable error whilst processing request.", response = ErrorResponse.class)
-            })
+        value = {
+            @ApiResponse(code = 400, message = "Invalid request", response = ErrorResponse.class),
+            @ApiResponse(code = 409, message = "Multiple offenders found in the same state ", response = ErrorResponse.class)
+        })
     @GetMapping(path = "/offenders/nomsNumber/{nomsNumber}")
-    public ResponseEntity<OffenderDetailSummary> getOffenderDetails(
-            @ApiParam(name = "nomsNumber", value = "Nomis number for the offender", example = "G9542VP", required = true)
-            @NotNull @PathVariable(value = "nomsNumber") final String nomsNumber) {
-        final var offender = offenderService.getOffenderSummaryByNomsNumber(nomsNumber);
-        return offender.map(
-                offenderDetail -> new ResponseEntity<>(offenderDetail, OK)).orElse(new ResponseEntity<>(OffenderDetailSummary.builder().build(), NOT_FOUND));
+    public OffenderDetailSummary getOffenderDetails(
+        @ApiParam(name = "nomsNumber", value = "Nomis number for the offender", example = "G9542VP", required = true)
+        @NotNull @PathVariable(value = "nomsNumber") final String nomsNumber,
+        @ApiParam(name = "failOnDuplicate", value = "Should fail if multiple offenders found regardless of status", example = "true", defaultValue = "false")
+        final @RequestParam(value = "failOnDuplicate", defaultValue = "false") boolean failOnDuplicate
+    ) {
+        final Optional<OffenderDetailSummary> offender;
+        if (failOnDuplicate) {
+            offender = offenderService
+                .getSingleOffenderSummaryByNomsNumber(nomsNumber)
+                .getOrElseThrow(error -> new ConflictingRequestException(error.getMessage()));
+
+        } else {
+            offender = offenderService
+                .getMostLikelyOffenderSummaryByNomsNumber(nomsNumber)
```

Comment:
So I guess this could error if neither offender is active?  Definitely no other errors that could happen?

Reply:
Didn't realise that getOrElseThrow comes from the Either

Reply:
correct